### PR TITLE
HibernateDomainExporter and inheritance with composite-id

### PR DESCRIPTION
**Pull request by vveloso**
Modifications required to produce compilable code as described in https://github.com/querydsl/querydsl/issues/1459 .

The change did not break existing unit tests.

### CODE REVIEW

**Comment by timowest** [comment]
@vveloso Thanks for the PR! Could you also add a test case that fails without your fix?

**Comment by timowest** [comment]
Also could you use a more descriptive commit message?

**Comment by vveloso** [comment]
I'm not sure what would be the best way to test this. I don't like to assert on text output, but it may well be what's required in this case. I'll take another look at the existing test cases to figure out how to go about building a new test case.

**Comment by Shredder121** [comment]
If you add your source that didn't compile at first (or at least the relevant parts) to the test sources, then it will be processed, if there is a regression it will not compile as before.
Does that sound like something that you can do?

**Comment by vveloso** [comment]
@Shredder121, I hope that this is what you meant. If not, it still seems to work: if the change I made to EntitySerializer is removed the Java code generated by the unit test will not compile.

**Comment by Shredder121** [comment]
> ... if the change I made to EntitySerializer is removed the Java code generated by the unit test will not compile

That is exactly what I meant, then if (for some reason) the logic is changed in such a way that the code doesn't compile anymore we'll know about it.

File: querydsl-jpa-codegen/src/test/java/com/mysema/query/jpa/domain18/Route1.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package com.mysema.query.jpa.domain18;
+
+public class Route1 {
+
+	private RouteDbKey routeKey;
```
Comment (timowest) [inline querydsl-jpa-codegen/src/test/java/com/mysema/query/jpa/domain18/Route1.java]: no tabs, only spaces

Comment (vveloso) [inline querydsl-jpa-codegen/src/test/java/com/mysema/query/jpa/domain18/Route1.java]: How many spaces?

Comment (timowest) [inline querydsl-jpa-codegen/src/test/java/com/mysema/query/jpa/domain18/Route1.java]: 4 spaces

Comment (vveloso) [inline querydsl-jpa-codegen/src/test/java/com/mysema/query/jpa/domain18/Route1.java]: Done.

File: querydsl-codegen/src/main/java/com/mysema/query/codegen/EntitySerializer.java [hunk 2]
```diff
@@ -64,18 +64,25 @@ public EntitySerializer(TypeMappings mappings, @Named("keywords") Collection<Str
         this.keywords = keywords;
     }
 
+	private boolean superTypeHasEntityFields(EntityType model) {
```
Comment (timowest) [inline querydsl-codegen/src/main/java/com/mysema/query/codegen/EntitySerializer.java]: indentation

File: querydsl-jpa-codegen/src/test/java/com/mysema/query/jpa/domain18/RouteDbKey.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package com.mysema.query.jpa.domain18;
+
+public class RouteDbKey {
+
+	private String type;
```
Comment (timowest) [inline querydsl-jpa-codegen/src/test/java/com/mysema/query/jpa/domain18/RouteDbKey.java]: tabs to spaces

**Comment by vveloso** [comment]
All spaces in the right places (hopefully).

**Comment by vveloso** [comment]
Sorry about the accidental close&reopen.

@timowest I hope that the latest commits address your concerns.

**Comment by timowest** [comment]
Looks good. Could you also provide a PR for the master branch?

**Comment by Shredder121** [comment]
To me it also looks good, so if the master counterpart could be created it is fine to merge.

**Comment by vveloso** [comment]
@timowest I just created the PR for the master branch (#1509).

**Comment by johnktims** [comment]
Looks good to me. Thanks for the PR @vveloso!

**Pull request merged**